R/install.R

```diff
@@ -396,3 +398,172 @@ build_status_ok <- function(process_log, quiet = FALSE) {
+            file.path("c:/", "Rtools"),
+            file.path("c:/", "Rtools35"),
+            file.path("c:/", "RTools"),
+            file.path("c:/", "RTools35"),
+            file.path("c:/", "rtools"),
+            file.path("c:/", "rtools35")
+          )
+          for (p in typical_paths) {
+            if (dir.exists(p)) {
+              rtools_path <- p
+            }
+          }
+        }
+        if (nzchar(rtools_path)) {
+          if (!fix) {
+            error_message <- c(
+              "",
+              "A RTools installation was found, but the PATH is not properly set.",
+              "Run 'cmdstanr::check_cmdstan_toolchain(check_only = FALSE)' to fix the issue."
+            )
+            stop(paste0(error_message, collapse = "\n"), call. = FALSE)
+          }
+          if (!quiet) message("Writing RTools path to ~/.Renviron ...")
+          if (nzchar(Sys.getenv("RTOOLS35_HOME"))) {
+            write(paste0('RTOOLS35_HOME=', rtools_path), file = "~/.Renviron", append = TRUE)
+          }
+          write('PATH="${RTOOLS35_HOME}\\bin;${RTOOLS35_HOME}\\mingw_64\\bin;${PATH}"', file = "~/.Renviron", append = TRUE)
+          stop("Restart R and run check_cmdstan_toolchain(check_only = TRUE) to confirm the installation was successful.", call. = FALSE)
+        } else {
+          error_message <- c(
+            "",
+            "A toolchain was not found. Please install RTools 3.5 and run",
+            "",
+            "write(\'RTOOLS35_HOME=rtools35/install/path/\', file = \"~/.Renviron\", append = TRUE)",
+            "replacing 'rtools35/install/path/' with the actual install path of RTools 3.5.",
+            "",
+            "Then restart R and run 'cmdstanr::check_cmdstan_toolchain(check_only = FALSE)'."
+          )
+          stop(paste0(error_message, collapse = "\n"), call. = FALSE)
+        }
+      }
+    }
+  } else {
+    # On Unix systems we check for make and a suitable compiler
+    make_path <- dirname(Sys.which("make"))
+    if (!nzchar(make_path)) {
+      if (os_is_macos()) {
+        stop("The 'make' tool was not found. Please install 'xcode' toolchains with 'xcode-select --install' or the app store and restart the R session.", call. = FALSE)
+      } else {
+        stop("The 'make' tool was not found. Please install 'make' and restart the R session.", call. = FALSE)
+      }
+      
+    }
+    gpp_path <- dirname(Sys.which("g++"))
+    clang_path <- dirname(Sys.which("clang++"))
+    if (!nzchar(gpp_path) && !nzchar(clang_path)) {
+      if (os_is_macos()) {
+        stop("A suitable C++ compiler was not found. Please install 'xcode' toolchains with 'xcode-select --install' or the app store and restart the R session.", call. = FALSE)
+      } else {
+        stop("A C++ compiler was not found. Please install the 'clang++' or 'g++' compiler and restart the R session.", call. = FALSE)
+      }
+    }
+  }
+  if (!quiet) message("The CmdStan toolchain is setup properly!")
```

Comment:
Nitpicking for code style consistency: 

```suggestion
  if (!quiet) {
    message("The CmdStan toolchain is setup properly!")
  }
```